firedrake/slope_limiter/limiter.py

```diff
@@ -0,0 +1,121 @@
+from firedrake import *
+from abc import *
+
+class _Limiter(object):
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def apply(self, field):
+        return NotImplemented
+
+    @abstractmethod
+    def compute_bounds(self, field):
+        return NotImplemented
```

Comment:
We should discuss what the separate public-facing methods should be.  I think these two are reasonable.  Do we also need `compute_centroids`, `apply_limiter`?  Others?